### PR TITLE
Dougsch add tls

### PR DESCRIPTION
**Pull request by Doug-AWS**
*Issue #, if available:*
New topic on setting TLS to 1.2 in Go.

*Description of changes:*
New topic (tls) and additional blurb in data protection topic (with link to tls topic).

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: doc_source/tls.rst [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+
+#####################################
+Enforcing TLS Version 1.2 in |sdk-go|
+#####################################
+
+.. meta::
+   :description: Describes how to set the TLS version for the |sdk-go|.
+
+To add increased security when communicating with AWS services, you should configure your client to use TLS 1.2 or later.
+TLS 1.2 has been the default in Go since release 1.13 in September 2019.
```
Comment (jasdel) [inline doc_source/tls.rst]: I don't think `TLS 1.2` is the default in Go. The [Go 1.13](https://golang.org/doc/go1.13#library) release didn't mention this change, and the current Go documentation states TLS 1.0 as the default minimum version.

[doc](https://golang.org/pkg/crypto/tls/#Config.MinVersion)
```go
    // MinVersion contains the minimum TLS version that is acceptable.
    // If zero, TLS 1.0 is currently taken as the minimum.
    MinVersion uint16 // Go 1.2
```

Go did add standard support for TLS 1.3 in Go 1.13. [Go 1.14](https://golang.org/doc/go1.14#library) did remove support for SSL 3.0 though.

**Review by jasdel — CHANGES_REQUESTED** [review]

File: doc_source/tls.rst (hunk 1 above)
Comment (Doug-AWS) [reply doc_source/tls.rst]: I've removed the second sentence.

File: doc_source/data-protection.rst [hunk 2]
```diff
@@ -29,7 +29,7 @@ For data protection purposes, we recommend that you protect |AWS| account creden
 you secure your data in the following ways:
 
 * Use multi-factor authentication (MFA) with each account.
-* Use SSL/TLS to communicate with |AWS| resources.
+* Use SSL/TLS to communicate with |AWS| resources. See :doc:`tls` for details.
 * Set up API and user activity logging with |CTlong|.
 * Use |AWS| encryption solutions, along with all default security controls within |AWS| services.
```
Comment (beckandros) [inline doc_source/data-protection.rst:34]: edit: ... |AWS| encryption solutions, with all default...

File: doc_source/tls.rst [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+#####################################
+Enforcing TLS Version 1.2 in |sdk-go|
+#####################################
+
+.. meta::
+   :description: Describes how to set the TLS version for the |sdk-go|.
+
+To add increased security when communicating with AWS services, you should configure your client to use TLS 1.2 or later.
+
+How do I set my TLS Version?
```
Comment (beckandros) [inline doc_source/tls.rst:20]: title: ...TLS version (lowercase)

File: doc_source/tls.rst [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+============================
+
+You can set the TLS version to 1.2 using the following code.
+
+1. Create a custom HTTP transport to require a minimum TLS 1.2 version.
```
Comment (beckandros) [inline doc_source/tls.rst:25]: ... require a minimum version of TLS 1.2

File: doc_source/tls.rst [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+             
+       // In Go versions before 1.13
+       err := http2.ConfigureTransport(tr)
+       if err != nil {
+           fmt.Println("Got an error configuring HTTP transport")
+           fmt.Println(err)
+           return
+       }
+
+       // In Go versions from 1.13 on:
```
Comment (beckandros) [inline doc_source/tls.rst:47]: ...versions later than 1.13 on 
(and no ending colon)

File: doc_source/tls.rst [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+           TLSClientConfig: &tls.Config{
+               MinVersion: tls.VersionTLS12,
+           },
+       }
+
+2. Configure the transport.
+
+   .. code:: go
+             
+       // In Go versions before 1.13
```
Comment (beckandros) [inline doc_source/tls.rst:39]: In Go versions earlier than 1.13

**Review by beckandros — COMMENTED** [review]
Small edits.

**Review by jasdel — APPROVED** [review]

File: doc_source/data-protection.rst (hunk 2 above)
Comment (Doug-AWS) [reply doc_source/data-protection.rst:34]: Done.

File: doc_source/tls.rst (hunk 3 above)
Comment (Doug-AWS) [reply doc_source/tls.rst:20]: Done

File: doc_source/tls.rst (hunk 4 above)
Comment (Doug-AWS) [reply doc_source/tls.rst:25]: Done

File: doc_source/tls.rst (hunk 5 above)
Comment (Doug-AWS) [reply doc_source/tls.rst:47]: Done

File: doc_source/tls.rst (hunk 6 above)
Comment (Doug-AWS) [reply doc_source/tls.rst:39]: Done

**Pull request merged**